tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/Emitter.scala

```diff
@@ -244,6 +282,7 @@ final class Emitter private (semantics: Semantics, outputMode: OutputMode,
 
     def getCache(version: Option[String]): DesugaredClassCache = {
       if (_cache == null || _lastVersion.isEmpty || _lastVersion != version) {
+        invalidate()
```

Comment:
Ah, I didn't think about this. Probably not a performance problem if it doesn't deregister itself.